Upload binaries during CI build (#374)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/config.md

| `OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES` | Sets the filename of executables the profiler cannot attach to. If not defined (default), the profiler will attach to any process. Supports multiple values separated with comma, for example: `MyApp.exe,dotnet.exe` |  |
| `OTEL_DOTNET_AUTO_AZURE_APP_SERVICES` | Set to indicate that the profiler is running in the context of Azure App Services. | `false` |

> ## Resource

Author: Read more: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md

It is explained just below 😄

> ## Resource

Reviewer: Kind of a trick question; I know it's a spec entity, but then I'd say "Resources" instead, as you may have more than one.

> ## Resource

Author: ```suggestion
## Resources
```

> ## Resource

Reviewer: +1

| `OTEL_DOTNET_AUTO_CLR_ENABLE_INLINING` | Set to `false` to disable JIT inlining. | `true` |
| `OTEL_DOTNET_AUTO_CLR_ENABLE_NGEN` | Set to `false` to disable NGEN images. | `true` |

## ASP.NET (.NET Framework) Instrumentation

> ASP.NET instrumentation on .NET Framework requires installing the
> [`OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule`](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/)
> NuGet package on the instrumented project.
> More info [here](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNet#step-2-modify-webconfig).

Reviewer: ```suggestion
ASP.NET instrumentation on .NET Framework requires installing the
[`OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule`](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/)
NuGet package on the instrumented project.
See [the WebConfig section](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNet#step-2-modify-webconfig) for more information.
```

  E.g. by adding `<PackageReference Include="Grpc.Net.Client" Version="2.32.0" />` to the `.csproj` file.
- On .NET Framework, using the `grpc` OTLP exporter protocol is not supported.
  
## Batch Span Processor

> The Batch Span Processor batches of finished spans before they are send by the exporter.

Reviewer: ```suggestion
The Batch Span Processor batches finished spans before they are sent by the exporter.
```

The plugin must use the same version of the `OpenTelemetry` as the
OpenTelemetry .NET AutoInstrumentation. Current version is `1.2.0-rc2`.

## .NET CLR Profiler

> OpenTelemetry .NET Auto-Instrumentation has to be configured
> as .NET CLR Profiler in order to peform bytecode instrumentation.

Reviewer: ```suggestion
 To perform bytecode instrumentation, configure the OpenTelemetry .NET
 auto-instrumentation as a .NET CLR Profiler.
```

#### .NET Core Dependency File

To fix assembly version conflicts in .NET Core, the `<application>.deps.json`, generated by default
during the build of .NET Core applications, needs to be modified. Build a .NET Core app with package
references using the required version and use the respective `deps.json` file to see what changes
are needed.

To experiment with modifying the `deps.json` file, add a reference to the required version OpenTelemetry
package to the [samples/CoreAppOldReference](./samples/CoreAppOldReference/) sample and rebuild the
application. Save the generated `deps.json` file, remove the package reference and rebuild the
sample app. Compare the files to understand the changes.

### Linux instrumentation not working

The proper binary needs to be selected when deploying to Linux, eg.: the default Microsoft .NET images are
based on Debian and should use the `deb` package, see the [Linux](#Linux) setup section.

If you are not sure what is the Linux distribution being used try the following commands:

```terminal
lsb_release -a
cat /etc/*release
cat /etc/issue*
cat /proc/version

> ```

Author: this section is removed - it was invalid (we do not build any packages)

### High CPU usage

> The default installation of auto-instrumentation enables tracing all .NET processes on the box.

Author: this line was not true. it has been fixed

Reference: [.NET Runtime Profiler Loading](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/profiling/Profiler%20Loading.md).

## .NET Runtime Additional-Deps and Package Store

To resolve assembly version conflicts in .NET Core,

> it is advised to set the

Reviewer: ```suggestion
set the
```

[`DOTNET_ADDITIONAL_DEPS`](https://github.com/dotnet/runtime/blob/main/docs/design/features/additional-deps.md)
and [`DOTNET_SHARED_STORE`](https://docs.microsoft.com/en-us/dotnet/core/deploying/runtime-store)

> environment variables to the values described below.

Reviewer: ```suggestion
environment variables to the following values:
```
